src/main/java/org/candlepin/insights/inventory/InventoryService.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2009 - 2016 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.inventory.client;
+
+import org.candlepin.insights.exceptions.RhsmConduitException;
+import org.candlepin.insights.inventory.client.model.FactSet;
+import org.candlepin.insights.inventory.client.model.Host;
+import org.candlepin.insights.inventory.client.model.HostOut;
+import org.candlepin.insights.inventory.client.resources.HostsApi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * A wrapper for the insights inventory client.
+ */
+@Component
+public class InventoryService {
+
+    private static Logger log = LoggerFactory.getLogger(InventoryService.class);
+
+    private HostsApi hostsInventoryApi;
+
+    @Autowired
+    public InventoryService(ApiClientFactory clientFactory) throws Exception {
+        hostsInventoryApi = new HostsApi(clientFactory.getObject());
+    }
+
+    // TODO Update the parameters once we know what will be coming from the RHSM service.
+    public HostOut sendHostUpdate(String orgId, String displayName, String hostName, UUID rhsmUuid)
+        throws RhsmConduitException {
+        try {
+            return hostsInventoryApi.apiHostAddHost(forgeAuthHeader(orgId),
+                createHost(orgId, displayName, hostName, rhsmUuid));
+        }
+        catch (Exception e) {
+            // FIXME This should all get removed once the exception mappers are in place.
+            e.printStackTrace();
```

Comment:
Please log the exception instead, even if its only temporary.

Reply:
DONE - Will be included in next commit.